test.js

```diff
@@ -223,3 +224,35 @@ test('iterator with byte range', function(t){
     t.end()
   })
 })
+
+test('backing rbtree is buffer-aware', function(t){
+  var db = new MemDOWN()
+    , noerr = function (err) {
+      t.error(err, 'opens correctly')
+    }
+    , noop = function () {}
+    , iterator
+
+  db.open(noerr)
+
+  var one = new Buffer('80', 'hex' )
+  var two = new Buffer('c0', 'hex' )
+
+  t.ok(two.toString() === one.toString(), 'would be equal when not buffer-aware')
+  t.ok(ltgt.compare(two, one) > 0, 'but greater when buffer-aware')
+
+  db.put(one, 'one', noop)
+
+  db.get(one, { asBuffer: false }, function(err, value) {
```

Comment:
this should be in the callback to `put()`; the fact that it works is due to Zalgo-ish behavior on our part. I can fix that on the way in, though; no worries :)